framework/src/wifi_manager/wifi_manager.c

```diff
@@ -654,29 +666,78 @@ wifi_manager_result_e _start_dhcpd(void)
 		WIFIADD_ERR_RECORD(ERR_WIFIMGR_SOFTAP_DHCPD_FAIL);
 	}
 	return wret;
+#else
+	struct netif *cur_netif;
+	cur_netif = netif_find(WIFIMGR_SOFTAP_IFNAME);
+	if (cur_netif == NULL) {
+		ndbg("[WM] DHCP Server - no network interface for dhcpd start\n");
+		WIFIADD_ERR_RECORD(ERR_WIFIMGR_CONNECT_DHCPD_FAIL);
+		return wret;
+	}
+	netif_set_up(cur_netif);
+	if (dhcps_start(cur_netif, _wifi_dhcpd_event) == ERR_OK) {
+		ndbg("[WM} DHCP Server - started successfully (lwip)\n");
+#ifdef CONFIG_ENABLE_IOTIVITY
+		__tizenrt_manual_linkset("gen");
+#endif
+		wret = WIFI_MANAGER_SUCCESS;
+	}
+	return wret;
+
+#endif
 }
 
 
 wifi_manager_result_e _stop_dhcpd(void)
 {
 	struct in_addr in = { .s_addr = INADDR_NONE };
 	WIFIMGR_SET_IP4ADDR(WIFIMGR_SOFTAP_IFNAME, in, in, in);
-
+	wifi_manager_result_e wret = WIFI_MANAGER_FAIL;
+#ifndef CONFIG_WIFIMGR_ENABLE_LWIP_DHCP
 	dhcpd_stop();
-
 #ifdef CONFIG_ENABLE_IOTIVITY
 	__tizenrt_manual_linkset("del");
 #endif
-	return WIFI_UTILS_SUCCESS;
+	wret = WIFI_MANAGER_SUCCESS;
+	return wret;
+#else
+	struct netif *cur_netif;
+	cur_netif = netif_find(WIFIMGR_SOFTAP_IFNAME);
+	if (cur_netif == NULL) {
+		ndbg("[WM] DHCP Server - no network interface for dhcpd stop\n");
+		WIFIADD_ERR_RECORD(ERR_WIFIMGR_CONNECT_DHCPD_FAIL);
+		return wret;
+	}
+	netif_set_down(cur_netif);
+	if (cur_netif->dhcps_pcb == NULL) {
+		WIFIADD_ERR_RECORD(ERR_WIFIMGR_CONNECT_DHCPD_FAIL);
+		ndbg("[WM] DHCP Server - stop dhcpd fail: no pcb\n");
```

Comment:
Why is WIFI_MANAGER_SUCCESS returned if DHCP server stop fails?

Reply:
Both dhcps_stop and dhcpd_stop have a void return type, because they perform a simple action that only chage the state and release the pointers, if exist. Therefore, DHCP server stop will not fail, (or failure will be handled by itself), and thus, this function returns success after calling dhcpd(s) stop function. Thanks for the comment.